### PR TITLE
Override IsByRefLike on S.R.E.*Builder classes

### LINKED ISSUE #8828: Override IsByRefLike property getter on RefEmit.***Builder classes.
They can just return false - this is for consistency with how other RefEmit Type-derived classes handle the Is predicates.

### PR DESCRIPTION
**Pull request by marcusturewicz**
Fixes #8828

### CODE REVIEW

File: src/coreclr/src/System.Private.CoreLib/src/System/Reflection/Emit/EnumBuilder.cs [hunk 1]
```diff
@@ -91,6 +91,8 @@ public FieldBuilder DefineLiteral(string literalName, object? literalValue)
 
         public override Type? BaseType => m_typeBuilder.BaseType;
 
+        public override bool IsByRefLike => false;
```
Comment (stephentoub) [inline src/coreclr/src/System.Private.CoreLib/src/System/Reflection/Emit/EnumBuilder.cs:94]: Is it the case that all of these throw NotSupportedException today and this PR is just making them return false instead?

**Comment by stephentoub** [comment]
Thanks for the PR.  Can you please add tests to go along with this?

File: src/coreclr/src/System.Private.CoreLib/src/System/Reflection/Emit/EnumBuilder.cs (hunk 1 above)
Comment (marcusturewicz) [reply src/coreclr/src/System.Private.CoreLib/src/System/Reflection/Emit/EnumBuilder.cs:94]: That's correct:

```
System.Reflection.Emit.Tests.EnumBuilderPropertyTests.IsByRefLikeFalse [FAIL]
        System.NotSupportedException : Derived classes must provide an implementation.
        Stack Trace:
          /dotnet/runtime/src/libraries/System.Private.CoreLib/src/System/Type.cs(54,0): at System.Type.get_IsByRefLike()
          /dotnet/runtime/src/libraries/System.Reflection.Emit/tests/EnumBuilder/EnumBuilder.Properties.Tests.cs(63,0): at System.Reflection.Emit.Tests.EnumBuilderPropertyTests.IsByRefLikeFalse()
```

Sorry, I forgot tests; adding them now.

**Review by stephentoub — APPROVED** [review]

**Comment by stephentoub** [comment]
The test failures are valid and are because a similar change needs to be made in mono's corelib.

**Comment by marcusturewicz** [comment]
> The test failures are valid and are because a similar change needs to be made in mono's corelib.

Ahh yes, there's two runtimes now hey. I'll fix that

**Review by marek-safar — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]
Thanks.

**Pull request merged**